File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        ref = {}
        for i in words[0]:
            ref[i] = ref.get(i, 0) + 1
        for j in range(1, len(words)):
            w = words[j]
            ref2 = {}
            for k in w:
                ref2[k] = ref2.get(k, 0) + 1
            for key, val in ref.items():
                if key in ref2:
                    ref[key] = min(val, ref2[key])
                else:
                    ref[key] = 0
        ans = []
        for key, val in ref.items():
            while val >= 1:
                ans.append(key)
                val -= 1
        return ans
                

File: 1002-find-common-characters/1002-find-common-characters.go
func commonChars(words []string) []string {
    ref := make([]int, 26)
    for _, char := range words[0] {
        ref[char-'a']++
    }
    var ans []string
    for i := 1; i < len(words); i++ {
        ref2 := make([]int, 26)
        w := words[i]
        for _, char := range w {
            ref2[char-'a']++
        }
        for idx, val := range ref {
            if val > ref2[idx] {
                ref[idx] = ref2[idx]
            }
        }
    }
    for idx, val := range ref {
        for val > 0 {
            ans = append(ans, string('a' + idx))
            val--
        }
    }
    return ans
}